R/FilterStates.R

```diff
@@ -184,34 +188,20 @@ FilterStates <- R6::R6Class( # nolint
       "subset"
     },
 
-    #' @description
-    #' Gets the number of active `FilterState` objects in this `FilterStates`.
-    #' @return `integer(1)`
-    get_filter_count = function() {
-      sum(vapply(private$queue, function(queue) {
-        queue$size()
-      }, FUN.VALUE = integer(1)))
-    },
+    # queue methods ----
 
     #' @description
-    #' Remove all `FilterState` objects from all queues in this `FilterStates`.
-    #' @return NULL
-    queue_empty = function() {
-      logger::log_trace("{ class(self)[1] } emptying queue, dataname: { deparse1(private$input_dataname) }")
-      queue_indices <- if (is.null(names(private$queue))) {
-        seq_along(private$queue)
-      } else {
-        names(private$queue)
-      }
-
-      lapply(queue_indices, function(queue_index) {
-        queue_elements <- names(self$queue_get(queue_index = queue_index))
-        lapply(queue_elements, function(element_id) {
-          self$queue_remove(queue_index = queue_index, element_id = element_id)
-        })
-      })
-
-      logger::log_trace("{ class(self)[1] } emptied queue, dataname: { deparse1(private$input_dataname) }")
+    #' Sets `ReactiveQueue` objects.
+    #' @param x (`list` of `ReactiveQueue`)\cr
+    #'  must be a list even if single `ReactiveQueue` is set.
+    # queue_initialize = function(x) {
+    #   checkmate::assert_list(x, types = "ReactiveQueue", min.len = 1)
+    #   private$queue <- x
+    #   invisible(NULL)
+    # },
+    queue_initialize = function(x) {
+      checkmate::assert_list(x, min.len = 1)
+      private$queue <- x
```

Comment:
I think this is confusing as it's not reactiveVal, but it is in all child classes. Maybe:
```suggestion
    queue_initialize = function(x) {
      checkmate::assert_list(x, min.len = 1)
      private$queue(x)
```
?

Reply:
You mean change `private$queue` (renamed) in `FilterStates` from a list of `reactiveVal`s to a `reactiveVal` containing such a list?

Reply:
ah - ok I misunderstood here I was confusing this function with the calls like this:

```
self$queue_initialize(
        list(
          y = reactiveVal()
        )
      )
```

I wonder if we actually need this `queue_initialize` function if we can just set it in the initializers directly

Reply:
I support @nikolas-burkoff idea in this case